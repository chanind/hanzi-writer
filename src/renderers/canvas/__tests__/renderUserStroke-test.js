const renderUserStroke = require('../renderUserStroke');

describe('renderUserStroke', () => {

  let ctx;

  beforeEach(() => {
    ctx = document.createElement('canvas').getContext('2d');
  });

  it('renders a user stroke path', () => {
    const props = {
      strokeColor: {r: 12, g: 101, b: 20, a: 0.3},
      strokeWidth: 2,
      opacity: 0.7,
      points: [{ x: 0, y: 0}, { x: 17, y: 3 }, { x: 9, y: 18 }],
    };
    renderUserStroke(ctx, props);
    expect(ctx.__getEvents()).toMatchSnapshot();
  });

  it('skips rendering if opacity is close to 0', () => {
    const props = {
      strokeColor: {r: 12, g: 101, b: 20, a: 0.3},
      strokeWidth: 2,
      opacity: 0.01,
      points: [{ x: 0, y: 0}, { x: 17, y: 3 }, { x: 9, y: 18 }],
    };
    renderUserStroke(ctx, props);
    expect(ctx.__getEvents()).toEqual([]);
  });

});
