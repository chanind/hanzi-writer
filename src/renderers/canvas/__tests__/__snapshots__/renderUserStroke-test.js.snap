// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderUserStroke renders a user stroke path 1`] = `
Array [
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "save",
  },
  Object {
    "props": Object {
      "value": 0.7,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "globalAlpha",
  },
  Object {
    "props": Object {
      "value": 2,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  Object {
    "props": Object {
      "value": "rgba(12, 101, 20, 0.3)",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": "round",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineCap",
  },
  Object {
    "props": Object {
      "value": "round",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineJoin",
  },
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 0,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  Object {
    "props": Object {
      "x": 17,
      "y": 3,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "x": 9,
      "y": 18,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 17,
            "y": 3,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
        Object {
          "props": Object {
            "x": 9,
            "y": 18,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "restore",
  },
]
`;
